### ISSUE TITLE
training data set

### ISSUE THREAD

**Issue by kshitij91** [tensorflow/ranking#133]
Hi team, 

I tried running the example notebook for handling sparse data, it runs fine. When I am running it on my dataset in ELWC format it gives me the below error:

Training with estimator made no steps. Perhaps input is empty or misspecified.

As a result of the same, no steps are getting executed, my example ELWC format is :

examples {
  features {
    feature {
      key: "attribute"
      value {
        bytes_list {
          value: "user"
          value: "type"
        }
      }
    }
    feature {
      key: "counts"
      value {
        int64_list {
          value: 979
        }
      }
    }
  }
}
examples {
  features {
    feature {
      key: "attribute"
      value {
        bytes_list {
          value: "p"
          value: "range"
        }
      }
    }
    feature {
      key: "counts"
      value {
        int64_list {
          value: 163
        }
      }
    }
  }
}
examples {
  features {
    feature {
      key: "attribute"
      value {
        bytes_list {
          value: "dummy"
        }
      }
    }
    feature {
      key: "counts"
      value {
        int64_list {
          value: 0
        }
      }
    }
  }
}
examples {
  features {
    feature {
      key: "attribute"
      value {
        bytes_list {
          value: "dummy"
        }
      }
    }
    feature {
      key: "counts"
      value {
        int64_list {
          value: 0
        }
      }
    }
  }
}
context {
  features {
    feature {
      key: "Query"
      value {
        bytes_list {
          value: "shoes"
        }
      }
    }
  }
}
 
keeping list size as 5
and further writing it to a file using tf record writer
am i doing something wrong or missing anything here?

**Comment by ramakumar1729** [comment]
@kshitij91 : Can you share your modified colab code, preferably via github gist? Please check if you have modified your feature columns to reflect your data.

**Comment by kshitij91** [comment]
https://gist.github.com/kshitij91/acd6789567a3af308a5a9d3469eacf95

I think I am missing somethin but not able to figure out, I have changed_LABEL_FEATURE from relevance to counts as per my data.

I am using your handling_sparse_features example for my POC

**Comment by xuanhuiwang** [comment]
@kshitij91, how does your tfrecord file name  look like?

Also, you may try out the script version here: https://github.com/tensorflow/ranking/blob/master/tensorflow_ranking/examples/tf_ranking_tfrecord.py

**Comment by kshitij91** [comment]
@xuanhuiwang
 _TRAIN_DATA_PATH = "customers_train.tfrecords"
_TEST_DATA_PATH = "customers_test.tfrecords"

**Comment by kshitij91** [comment]
@ramakumar1729 @xuanhuiwang  Also when I am running the given example with ANTIQUE: A Question Answering Dataset it executes steps and loss values are visible at training time also metrics get computed but when i am running the same example with example data given in data/train_elwc.tfrecords and data/eval_elwc.tfrecords, no loss values are seen and steps are also not executed in terminal it says "Training with estimator made no steps. Perhaps input is empty or misspecified."

Is it the expected behaviour? is it due to very less training data? not able to understand why it does not execute steps??

**Comment by xuanhuiwang** [comment]
I didn't see anything wrong here. It is probably your path is not specified correctly. Can you please try to use absolute path, instead of the relative ones?

**Comment by kshitij91** [comment]
@xuanhuiwang tried both the paths and path is correct as I can see number of tokens in vocab file in console, also i have tried reading the files separetely.

I am not able to understand why it is giving me "Training with estimator made no steps. Perhaps input is empty or misspecified." 
I1114 12:11:05.123650 4431357376 estimator.py:371] Loss for final step: None.

Is it due to less training data? when i run using antique dataset, same script shows proper steps and metrics but not with the example ELWC data given in data folder

**Comment by xuanhuiwang** [comment]
@kshitij91, I don't think the volume of the training data is a problem. I tried to use `data/eval_elwc.tfrecords` and `data/train_elwc.tfrecords` on my side, it trains fine. However, when I intentionally mis-specify the file name, I see the same error as you.

**Comment by kshitij91** [comment]
@xuanhuiwang @ramakumar1729 can you please share how did you create data/eval_elwc.tfrecords records from a text file or any other source?

**Comment by xuanhuiwang** [comment]
You may find something here: https://github.com/tensorflow/ranking/blob/master/tensorflow_ranking/examples/tf_ranking_tfrecord_test.py#L85.

**Comment by kshitij91** [comment]
@xuanhuiwang okay, i think i am clear now, thanks for help, closing the issue
